From 4e4743d9a09fc33e7fa06f16ee72b67c9fe6473a Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Tue, 30 Nov 2021 17:57:18 +0200
Subject: [PATCH] [SW-36213] preboot: include hdr with __packed str

As a preparation for adding packed struct in hl_fw_if,
it is required to add include for the __packed type header.

Change-Id: I94b753fdc8eddf50779ef87d08683173454ecb44
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 02798a4..32824c5 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -9,10 +9,10 @@
 #include <psoc_global_conf_regs.h>
 #include <pcie_aux_regs.h>
 #include <cpu_ca53_cfg_regs.h>
-#include <hl_fw_if.h>
 #include <hl_boot_if.h>
 #include <bitops.h>
 #include <preboot.h>
+#include <hl_fw_if.h> /* requires preboot.h for __packed */
 
 #include "common.h"
 #include "uart.h"
diff --git a/lib/common/module_i2c_slave.c b/lib/common/module_i2c_slave.c
index 577c372..a29885a 100644
--- a/lib/common/module_i2c_slave.c
+++ b/lib/common/module_i2c_slave.c
@@ -10,9 +10,9 @@
 #include <psoc_global_conf_regs.h>
 #include <psoc_i2c_s_regs.h>
 #include <psoc_i2c_s_masks.h>
-#include <hl_fw_if.h>
 #include <pcie_dbi_regs.h>
 #include <preboot.h>
+#include <hl_fw_if.h> /* requires preboot.h for __packed */
 #include <preboot/thermal.h>
 #include <hl_i2c_slave_events.h>
 #include <crc8.h>
diff --git a/lib/preboot/module_lkd.c b/lib/preboot/module_lkd.c
index 61efb91..7b7e1fe 100644
--- a/lib/preboot/module_lkd.c
+++ b/lib/preboot/module_lkd.c
@@ -6,8 +6,8 @@
 
 #include <string.h>
 #include <hl_boot_if.h>
-#include <hl_fw_if.h>
 #include <preboot.h>
+#include <hl_fw_if.h> /* requires preboot.h for __packed */
 #include <boot_if.h>
 #include <boot_info.h>
 #include <bitops.h>
